### ISSUE TITLE
Linux version

### ISSUE THREAD

**Issue by y2k04** [sandydoo/flux-screensavers#1]
Not sure if anyone actually wants it, but just in case anyone does want a Linux version, @sandydoo this might help http://www.diag.uniroma1.it//~liberato/screensaver

**Comment by sandydoo** [comment]
Thanks, @y2k04. A few people have actually requested a Linux version.
There are a few annoying bits I have yet to figure out:

- [ ] Better multi-monitor support.
	xscreensaver manages window creation for us and creates one window per monitor. I’d like to figure out a way to support drawing Flux seamlessly across multiple displays.
- [ ] (Optional) one-click install to add Flux to the list of xscreensaver hacks.
- [ ] Wayland. Not sure what the options are here.

**Comment by y2k04** [comment]
I'm going to build the latest version of your screensaver for linux and see what I can do

**Comment by y2k04** [comment]
I don't know how to fix this lol
https://github.com/y2k04/flux-screensavers/runs/7484308881

**Comment by sandydoo** [comment]
Yeah, the ffi crate will [only compile for macOS](https://github.com/sandydoo/flux-screensavers/blob/89b662fd9076d37d14220ac9a1ee5218d014d119/flux-ffi/src/lib.rs#L86-L87). It’s single purpose is to interface with Flux from Swift. What you actually need is the Windows version, but it’ll take some rewriting to get it to even compile for Linux. 

If you’re looking for something fun to hack on, you could modify `flux-desktop` in the [main repo](https://github.com/sandydoo/flux) and turn it into a Linux screensaver. You can borrow some of the window handling logic from [this old branch](https://github.com/sandydoo/flux-screensavers/blob/linux/init/linux/src/main.rs), stick it into `flux-desktop`, and it should™️  work.

**Comment by y2k04** [comment]
Not sure whether or not it worked, but it built successfully: https://github.com/y2k04/flux/actions/runs/3072065512

**Comment by y2k04** [comment]
Yeah no. This is what happens when I try running 'build-script-build' in Ubuntu WSL:
```
thread 'main' panicked at 'missing output directory', flux/build.rs:6:42
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

**Comment by y2k04** [comment]
Oh lol I was dumb when I made that comment. It literally said "missing output directory" 😂

**Comment by lexterror** [comment]
Could someone please provide the linux binaries? It would be really appreciated. Thank you!

**Comment by y2k04** [comment]
> Could someone please provide the linux binaries? It would be really appreciated. Thank you!

No such thing exists because we are still stuck trying to make it. It used to exist, but after things were rewritten there was no linux implementation after that. To be honest, I should look into XScreensaver again or whatever Linux/Ubuntu uses these days since I finished high school now.

**Comment by lexterror** [comment]
I dont want to sound weird! Are you saying its in xscreensavers now?

**Comment by y2k04** [comment]
No it's not. I meant the backend for screensavers on Linux.

**Comment by Sad-Soul-Eater** [comment]
> Not sure whether or not it worked, but it built successfully: [y2k04/flux/actions/runs/3072065512](https://github.com/y2k04/flux/actions/runs/3072065512)

I downloaded this artifact a few days ago, extract `target/x86_64-unknown-linux-gnu/release/flux-desktop`, add path to binary in ~/.xscreensaver, cursor visible on screensaver, but it works 🙃

https://user-images.githubusercontent.com/20245745/208187781-83c7626f-03b5-4b85-85f9-0dcd0fa65c62.mp4

**Comment by y2k04** [comment]
WAIT WHAT?!?!??!!?

**Comment by Sad-Soul-Eater** [comment]
To add flux-desktop to xscreeensaver edit `~/.xscreensaver`:
1. Add path to binary in program block (or if it already in $PATH - just binary name)
2. Change `selected` number to program index

![2022 12 17-12 57 31 screenshot](https://user-images.githubusercontent.com/20245745/208238483-0912acad-ce2b-432d-881a-5f3e4e55c410.jpg)

Also, I added next rule in my i3 config:
`for_window [class="flux-desktop"] fullscreen enable`

There is another issue, when you're selecting flux-desktop in xscreensaver-setting screensavers list, flux drawing in new window instead of settings window, probably because linux build doesn't realize some interface or idk 

![2022 12 17-13 14 24 screenshot](https://user-images.githubusercontent.com/20245745/208239018-43f9eda6-96e6-4494-bbf5-0694fd792f8d.jpg)

**Comment by scolastico** [comment]
@Sad-Soul-Eater or @y2k04 does anyone of you still have a compiled linux version arround? The repo from @y2k04 isnt available anymore sadly.

**Comment by y2k04** [comment]
> @Sad-Soul-Eater or @y2k04 does anyone of you still have a compiled linux version arround? The repo from @y2k04 isnt available anymore sadly.

I'm rerunning the GitHub build action and I will post it to my repo's releases so it doesn't disappear.

**Comment by y2k04** [comment]
https://github.com/y2k04/flux/releases/tag/v4.4.0-linux-unofficial
